src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerRootViewModel.cs

```diff
@@ -36,7 +40,13 @@ public StackTraceExplorerRootViewModel(IThreadingContext threadingContext, Visua
             set => SetProperty(ref _selectedTab, value);
         }
 
-        public void AddNewTab(bool triggerPaste)
+        /// <summary>
+        /// Returns true if there's a tab that already matches the text
+        /// </summary>
+        public bool ContainsTab(string text)
+            => Tabs.Any(tab => tab.Content.ViewModel.Matches(text));
```

Comment:
Is this slow and/or does it need the UI thread? I wonder what the trade off here is, if you have a lot of tabs, versus just keeping the most recent clipboard text seen in memory, and rejecting that.

Reply:
It's pretty fast just because it's a hashcode comparison. It doesn't need UI thread as long as the underlying `Tabs` isn't being modified...